File: Seori/Weekly Contest 414/3280. Convert Date to Binary.py
class Solution:
    def convertDateToBinary(self, date: str) -> str:
        # format() 함수에 'b'를 사용하면 2진수로 변환할 수 있다. You can convert to binary using 'b' in the format() function.
        year, month, day = map(int, date.split("-"))
        year = format(year, 'b')
        month = format(month, 'b')
        day = format(day, 'b')
        return year + '-' + month + '-' + day

File: Lucy/2024/Weekly Contest 414/3280_Convert_Date_to_Binary.js
/**
 * 3280. Convert Date to Binary
 *
 * url: https://leetcode.com/problems/convert-date-to-binary/description/
 * difficulty: Easy
 * topic: Math, String
 * date: 2024.09.17(TUE)
 */

/**
 * @param {string} date
 * @return {string}
 */
var convertDateToBinary = function (date) {
  const [year, month, day] = date.split("-").map(Number);
  return `${year.toString(2)}-${month.toString(2)}-${day.toString(2)}`;
};

console.log(convertDateToBinary("2080-02-29")); // "100000100000-10-11101"
console.log(convertDateToBinary("1900-01-01")); // "11101101100-1-1"
